From 24c7d508337bb12ab3ce62c4afe677d7817140ea Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Mon, 14 Jul 2025 16:23:17 +0200
Subject: [PATCH] Fix LINSTOR satellite LS_KEEP_RES regex

The original regex is not correct: luckily it matches resources like
`xcp-persistent-database` but also these resources:
`xcp-persisten`, `xcp-persistenttttttt`...

Also this issue has no real impact in production. It's just a typo.

After this fix we can still match patterns like "xcp-persistenttttttt" but
the goal here is to avoid detection of persistent volumes whose name contains
only the prefix `xcp-persisten` instead of `xcp-persistent`.

We could use more complex regexes, but I don't know where we're going with
SMAPIv3 and possible persistent volume features. So I don't prefer to use
more restrictive regexes for now.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 etc/systemd/system/linstor-satellite.service.d/override.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/systemd/system/linstor-satellite.service.d/override.conf b/etc/systemd/system/linstor-satellite.service.d/override.conf
index b1686b4f..f2d3893f 100644
--- a/etc/systemd/system/linstor-satellite.service.d/override.conf
+++ b/etc/systemd/system/linstor-satellite.service.d/override.conf
@@ -1,5 +1,5 @@
 [Service]
-Environment=LS_KEEP_RES=^xcp-persistent*
+Environment=LS_KEEP_RES=^xcp-persistent.*
 
 [Unit]
 After=drbd.service
